.github/workflows/common.yml

```diff
@@ -114,7 +114,7 @@ jobs:
           token: ${{ secrets.GITHUB_TOKEN }}
           interval: 45
           timeout: 1200
-          ref: ${{ github.event.pull_request.head.sha || github.event.push.head.sha }}
+          ref: ${{ github.event.pull_request.head.sha || github.sha }}
```

Comment:
In fact it seems there is no reason to run gatekeeper on pushes. May be it makes sense to extract it to the separate workflow in run it only for pull requests?

Reply:
I agree with you. I'd like even to expand the question. Do we really need to run tests against main branch? If I understand correctly, we cannot merge into main until PR tests are green.

Reply:
I tried to extract merge-gatekeeper into a separate flow, but it requires additional management to run it once one of the other jobs started.

Reply:
Since merge-gatekeeper is our exit point of all checks, and notifications are based on it, we have no options to skip it.